packit_service/worker/result.py

```diff
@@ -17,14 +17,13 @@ class TaskResults(dict):
 
     def __init__(self, success: bool, details: Dict[str, Any] = None):
         """
-
-        :param success: has the job handler succeeded:
-                          True - we processed the event
-                          False - there was an error while processing it -
-                                  usually an exception
-        :param details: more info from job handler
-                        (optional) 'msg' key contains a message
-                        more keys to be defined
+        Args:
+            success: Represents the resulting state of the job handler.
+                `True`, if we processed the event; `False` an error occurred
+                while processing it (usually an exception)
```

Comment:
@lbarcziova related to your #2526… Going through the Sentry we have a lot of “errors” that we can't do much about, e.g., failed SRPM build, failed RPM build; for both of the examples, there's rather small chance it is our fault, in case of SRPMs it could be (we run `packit srpm` in Copr), but it would have rather high impact for the Sentry error to be **the one** to raise the suspicion…

Since `success=False` implies `logger.error(some message)`, this causes a lot of entries in the Sentry. In the ideal world this should be an enumeration of `ok, nok, err` (or `success, fail, error`) where the `nok`/`fail` represents not successful result, but not directly failing on our side (e.g., failed (S)RPM build above), and `err`/`error` issue on our end (e.g. caught exception).

Additionally we could retry certain `nok`/`fail` (like we do right now for `sync-release`), but it doesn't make sense to retry an `error` which would be an internal error on our end.

---

Maybe open for a discussion on Thursday's arch.